### PR TITLE
Paginate through empty workflow history pages

### PR DESCRIPTION
**Pull request by drewhoskins-stripe**
# Description
Temporal server sometimes sends empty history pages but with a next_page_token.  Best practice across the SDKs is to keep paginating.

But temporal ruby was frequently returning a `Temporal::UnexpectedResponse`. This caused workflow tasks to fail and delay ten seconds until the next retry.  Especially prevalent on long workflow histories.

This change simply removes the check since this is supported behavior.

Update: I threw in a couple of unrelated test fixes to get the CircleCI tests to pass.

# Test plan

Note : I OKed this with Chad Retz from Temporal.  We've had this change active for months at Stripe and have been happy with it.

New test:
`rspec spec/unit/lib/temporal/workflow/task_processor_spec.rb`

### CODE REVIEW

File: examples/spec/integration/search_attributes_spec.rb [hunk 1]
```diff
@@ -47,24 +47,6 @@ def cleanup
     end.to raise_error(Temporal::SearchAttributeAlreadyExistsFailure)
   end
 
-  it 'type change fails' do
```
Comment (drewhoskins-stripe) [inline examples/spec/integration/search_attributes_spec.rb:50]: @jeffschoner-stripe, this worked fine in the circleci version and I figure it's not really important functionality to test-- like if temporal fixes it, everyone will be happy--so I just removed the test.

File: examples/bin/register_namespace [hunk 2]
```diff
@@ -13,4 +13,17 @@ rescue Temporal::NamespaceAlreadyExistsFailure
   Temporal.logger.info 'Namespace already exists', { namespace: namespace }
 end
 
-
+# Register a variety of search attributes for ease of integration testing
```
Comment (drewhoskins-stripe) [inline examples/bin/register_namespace:16]: fixes some tests that were broken in CI, I think based on some recent temporal docker image changes to stop registering these automatically.

Comment (jeffschoner-stripe) [reply examples/bin/register_namespace:16]: I think the behavior became inconsistent between how Temporal server is set up in temporalio/auto-setup (which the CI config here uses) and how it is setup in production and the docker-compose. The below code should handle both correctly, but Temporal should restore these to parity. I believe the auto-setup may be using SQL-based enhanced visibility where search attributes are scoped to a namespace and not global like with Elasticsearch.

File: examples/spec/integration/search_attributes_spec.rb (hunk 1 above)
Comment (jeffschoner-stripe) [reply examples/spec/integration/search_attributes_spec.rb:50]: I think this is another difference between SQL-based enhanced visibility and Elasticsearch-based visibility. I'm fine removing it for now, as we're mostly testing Temporal server, not temporal-ruby here.

**Review by jeffschoner-stripe — COMMENTED** [review]
@drewhoskins-stripe This looks good me. Did you want to fix the error where the Temporal Java test service returns a slightly different proto representation for an incomplete workflow history when awaiting results? It's tangentially related to this change, but I can also do it in a separate PR at some point.

**Review by DeRauk — APPROVED** [review]

**Pull request merged**